Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
-        DB.PlusOnes[playerName] = max(DB.PlusOnes[playerName] - 1, 0);
+        local TrustedPlayers = GL:strSplit(trustedPlayerCSV, ",");
+        tinsert(TrustedPlayers, playerName);
+        GL.Settings:set("PlusOnes.automaticallyAcceptDataFrom", table.concat(TrustedPlayers, ","));
     end
+end
 
-    self:triggerChangeEvent();
+--- Remove a player from the list of "trusted" players
+---
+---@param playerName string
+---@return void
+function PlusOnes:removePlayerFromTrusted(playerName)
+    playerName = strtrim(playerName);
+
+    -- No point removing the player if he's not trusted in the first place
+    if (GL:empty(playerName)
+        or not self:playerIsTrusted(playerName)
+    ) then
+        return;
+    end
+
+    local trustedPlayerCSV = GL.Settings:get("PlusOnes.automaticallyAcceptDataFrom", "");
+    local TrustedPlayers = GL:strSplit(trustedPlayerCSV, ",");
+    local NewTrustedPlayers = {};
+    local normalizedName = GL:normalizedName(playerName);
 
-    return DB.PlusOnes[playerName];
+    for _, trustedPlayer in pairs(TrustedPlayers) do
+        if (GL:normalizedName(trustedPlayer) ~= normalizedName) then
+            tinsert(NewTrustedPlayers, trustedPlayer);
+        end
+    end
+
+    GL.Settings:set("PlusOnes.automaticallyAcceptDataFrom", table.concat(NewTrustedPlayers, ","));
 end
 
---- Set a given player's plusOne value to 0
+--- Check whether plus ones are enabled
 ---
----@param playerName string
----@return number
-function PlusOnes:setToZero(playerName)
-    GL:debug("PlusOnes:setToZero");
+---@return boolean
+function PlusOnes:enabled()
+    GL:debug("PlusOnes:enabled");
+
+    return GL.Settings:get("PlusOnes.enabled", false);
+end
+
+--- Check whether there are plus ones available
+---
+---@return boolean
+function PlusOnes:available()
+    GL:debug("PlusOnes:available");
+
+    --return GL:higherThanZero(DB:get("PlusOnes.MetaData.updatedAt", 0)); --not sure this check is even needed for plus ones
+    return true;
+end
+
+--- Draw either the importer or overview
+--- based on the current boosted roll data
+---
+---@return void
+function PlusOnes:draw()
+    GL:debug("PlusOnes:draw");
+
+    -- Show the plus ones overview instead
+    GL.Interface.PlusOnes.Overview:draw();
+end
+
+--- Checks and handles whisper commands if enabled.
+---
+---@param _ string
+---@param message string
+---@param sender string
+---@return void
+function PlusOnes:handleWhisperCommand(_, message, sender)
+    GL:debug("PlusOnes:handleWhisperCommand");
+
+    -- Only listen to the following messages
+    if (not GL:strStartsWith(message, "!plusone")
+        and not GL:strStartsWith(message, "!PLUSONE")
+        and not GL:strStartsWith(message, "!po")
+        and not GL:strStartsWith(message, "!PO")
+        and not GL:strStartsWith(message, "!+1")
+    ) then
+        return;
+    end
+
+    local args = GL:strSplit(message, " ");
+
+    -- See if name is given.
+    if (#args > 1) then
```

Comment:
No need to count `args` if all you care about is `args[2]` being present.
```lua
if (args[2]) then
```
Would suffice

Reply:
I'll make the fix.